### PR TITLE
angular 4 upgrade / side-by-side

### PR DESCRIPTION
**Pull request by brendansudol**
this PR addresses https://github.com/18F/dol-whd-14c/issues/185

it adds the latest Angular framework (4) and bootstraps a hybrid application that allows for the existing Angular 1 components / app to continue working while supporting new components to be added using TypeScript and the latest Angular version

i primarily followed these two resources: [1](https://angular.io/guide/upgrade), [2](https://hackernoon.com/angular-v4-hybrid-upgrade-application-73d5afba1e01)

the existing app continues to function as it did before; in addition, per the acceptance criteria, i added a "hello world" Angular 4 component in TypeScript (see `src/v4/hello-world.component.ts`) that you can view at `https://localhost:3333/#!/v4/hello` (when running locally)

cc: @mmurthydol

### CODE REVIEW

File: DOL.WHD.Section14c.Web/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "sourceMap": true,
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "lib": ["es2015", "dom"],
+    "noImplicitAny": true,
```
Comment (mgwalker) [inline DOL.WHD.Section14c.Web/tsconfig.json]: I'm getting a number of red squigglies in Atom related to variables having an implicit `Any` type.  Is that related to this rule?  And if so, should those be addressed?

File: DOL.WHD.Section14c.Web/src/modules/app.js [hunk 2]
```diff
@@ -41,6 +43,13 @@ let app = angular.module('14c', [
   'pagination'
 ]);
 
+app.directive(
+  'helloWorld',
+  downgradeComponent({
+    component: HelloWorldComponent
+  })
+);
+
```
[the remark this answers is not in the document]
Comment (brendansudol) [reply DOL.WHD.Section14c.Web/src/modules/app.js]: exactly!

File: DOL.WHD.Section14c.Web/tsconfig.json (hunk 1 above)
Comment (brendansudol) [reply DOL.WHD.Section14c.Web/tsconfig.json]: hmmm, i'm not sure (don't have atom) -- let me change that rule and let me know if the red goes away

File: DOL.WHD.Section14c.Web/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "sourceMap": true,
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "lib": ["es2015", "dom"],
+    "noImplicitAny": false,
```
Comment (mgwalker) [inline DOL.WHD.Section14c.Web/tsconfig.json]: The squigglies went away. 🤷‍♂️  I did a little more googling and this is basically a compiler warning flag.  Maybe something to consider turning on again in the future to get the full power of type checking, but I think for this push, it's okay to turn off.

Comment (brendansudol) [reply DOL.WHD.Section14c.Web/tsconfig.json]: 💯

**Comment by mgwalker** [comment]
Also addresses #226
